activesupport/test/cache/behaviors/cache_store_behavior.rb

```diff
@@ -480,6 +480,27 @@ def test_expires_in
     end
   end
 
+  def test_expires_in_zero_means_infinite_expiry
+    time = Time.local(2008, 4, 24)
+
+    Time.stub(:now, time) do
+      @cache.write("foo", "bar", expires_in: 0)
+      assert_equal "bar", @cache.read("foo")
+    end
+
+    Time.stub(:now, time + 30) do
+      assert_equal "bar", @cache.read("foo")
+    end
+
+    Time.stub(:now, time + 61) do
+      assert_equal "bar", @cache.read("foo")
+    end
+
+    Time.stub(:now, time + 121) do
+      assert_equal "bar", @cache.read("foo")
+    end
```

Comment:
This suggestion is related to avoid flaky test related to the usage of `foo` on shared test. See more here: https://github.com/rails/rails/pull/43670
```suggestion
    time = Time.local(2008, 4, 24)
    key = SecureRandom.uuid
    value = SecureRandom.alphanumeric
    
    Time.stub(:now, time) do
      @cache.write(key, value, expires_in: 0)
      assert_equal value, @cache.read(key)
    end

    Time.stub(:now, time + 30) do
      assert_equal value, @cache.read(key)
    end

    Time.stub(:now, time + 61) do
      assert_equal value, @cache.read(key)
    end

    Time.stub(:now, time + 121) do
      assert_equal value, @cache.read(key)
    end
```